File: backend/src/main/java/com/org/playboard/dto/user/UserSummaryDto.java
package com.org.playboard.dto.user;

import com.org.playboard.entity.user.User;
import com.org.playboard.service.user.AvatarUrlResolver;
import java.util.List;
import java.util.UUID;

public record UserSummaryDto(
        UUID id,
        String displayName,
        String email,
        String photoUrl,
        String avatarId,
        String avatarColor,
        List<String> authProviders) {

    public static UserSummaryDto from(User user, AvatarUrlResolver avatarUrls, List<String> authProviders) {
        return new UserSummaryDto(
                user.getId(),
                user.getDisplayName(),
                user.getEmail(),
                avatarUrls.resolve(user.getPhotoUrl()),
                user.getAvatarId(),
                user.getAvatarColor(),
                List.copyOf(authProviders));
    }
}

File: android/app/src/main/java/com/org/playboard/data/remote/dto/UserSummaryDto.kt
package com.org.playboard.data.remote.dto

import kotlinx.serialization.Serializable

/** Mirrors the backend's {@code dto.user.UserSummaryDto}. */
@Serializable
data class UserSummaryDto(
    val id: String,
    val displayName: String,
    val email: String,
    val photoUrl: String?,
    val avatarColor: String,
)
